### PR TITLE
Multi device

### PR DESCRIPTION
**Pull request by dala318**
# What does this implement/fix?

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>
https://github.com/esphome/feature-requests/issues/1335 

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
device:
  - id: door_one
    name: Door one

binary_sensor:
  - platform: template
    device_id: door_one
    ...
```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/device/__init__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+MULTI_CONF = True
+
+CODEOWNERS = ["@dala318"]
+
+CONFIG_SCHEMA = cv.Schema(
+    {
+        # cv.Required(CONF_ID): cv.declare_id(DeviceClass),
+        cv.Required(CONF_ID): cv.declare_id(StringRef),
+        cv.Required(CONF_NAME): cv.string,
```
Comment (Unfocused) [inline esphome/components/device/__init__.py:17]: Should also allow specifying `area` too, like you can for the  main device config (added in [2023.11.0](https://esphome.io/changelog/2023.11.0#areas), see https://github.com/esphome/esphome/pull/5602). Home Assistant calls it `suggested_area`.

**Comment by Unfocused** [comment]
Devices in Home Assistant can specify a 'via_device' property, so Home Assistant can show how a device is related to another device. One of the use-cases is sub-devices specifying their parent device, which matches this perfectly.

**Comment by dala318** [comment]
> Devices in Home Assistant can specify a 'via_device' property, so Home Assistant can show how a device is related to another device. One of the use-cases is sub-devices specifying their parent device, which matches this perfectly.

Interesting, did not know about this but once I get to the HA integration part (whenever I find time to try that) it could make sense to add that.

**Comment by dala318** [comment]
Even if this could override some native behavior in some sense I don't think this is the correct way to do it. Closing this pull-request since I won't be able to finish it and have it be done in a proper way.

**Pull request closed without merging**